connexion/operation.py

```diff
@@ -195,11 +198,33 @@ def __init__(self, api, method, path, operation, resolver, app_produces, app_con
         self.path = path
         self.validator_map = dict(VALIDATOR_MAP)
         self.validator_map.update(validator_map or {})
-        self.security_definitions = security_definitions or {}
-        self.definitions = definitions or {}
-        self.parameter_definitions = parameter_definitions or {}
-        self.response_definitions = response_definitions or {}
+
+        # swagger2
+        self.definitions = definitions
+
+        self.security_definitions = security_definitions
+        self.parameter_definitions = parameter_definitions
+        self.response_definitions = response_definitions
+
+        # openapi3
+        self.components = components or {}
+
+        def component_get(oas3_name):
+            return self.components.get(oas3_name, {})
+
+        self.security_definitions = self.security_definitions or component_get('securitySchemes')
+        self.parameter_definitions = self.parameter_definitions or component_get('parameters')
+        self.response_definitions = self.response_definitions or component_get('responses')
+
         self.definitions_map = {
+            'components': {
```

Comment:
Q:  is this independent on self.*_definitions?

Reply:
the definitions map is used to resolve references, so its structure matches the "$ref" tags in people's specs (ex. `#/components/schemas/BreadFish"). In OAS3 these are all under components, but in Swagger2 there are some top level ones like "parameters", "definitions" and "responses".
https://blog.readme.io/an-example-filled-guide-to-swagger-3-2/ provides a nice visual representation of the differences.